includes/REST_API/Status_Check.php

```diff
@@ -0,0 +1,134 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\REST_API;
+
+use Google\Web_Stories\Decoder;
+use Google\Web_Stories\Experiments;
+use WP_REST_Server;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_Error;
+use WP_Post_Type;
+
+/**
+ * API endpoint check status.
+ *
+ * Class Status_Check
+ */
+class Status_Check extends WP_REST_Controller {
+	/**
+	 * Decoder instance.
+	 *
+	 * @var Decoder Decoder instance.
+	 */
+	private $decoder;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Experiments $experiments Experiments instance.
+	 */
+	public function __construct( Experiments $experiments ) {
+		$this->namespace = 'web-stories/v1';
+		$this->rest_base = 'status-check';
+		$this->decoder   = new Decoder( $experiments );
+	}
+
+	/**
+	 * Registers routes for links.
+	 *
+	 * @see register_rest_route()
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::ALLMETHODS,
+					'callback'            => [ $this, 'status_check' ],
+					'permission_callback' => [ $this, 'status_check_permissions_check' ],
+					'args'                => [
+						'content' => [
+							'description'       => __( 'Test HTML content.', 'web-stories' ),
+							'required'          => true,
+							'type'              => 'string',
+							'validate_callback' => [ $this, 'check_html_string' ],
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Status check, return true for now.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function status_check( $request ) {
+		$data = [
+			'success' => true,
+		];
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Checks if current user can process status.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function status_check_permissions_check() {
+		if ( ! current_user_can( 'edit_web-stories' ) ) {
+			return new WP_Error( 'rest_forbidden', __( 'Sorry, you are not allowed run status check.', 'web-stories' ), [ 'status' => rest_authorization_required_code() ] );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check that there is html in the string by removing html.
+	 *
+	 * @param string $param HTML string.
+	 *
+	 * @return bool
+	 */
+	public function check_html_string( $param ) {
```

Comment:
I don't think we need this validation callback.

If anything, the `status_check` method above could return the sent content or a hash of it

Reply:
This validates that the string submitted is HTML. This is test for firewalls that are removing html that is submitted. I can remove this check, but I think we need to check the values submitted to the api at PHP level, otherwise what is the point of submitting them?